x-pack/plugins/security/server/profile/profile_service.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, Logger } from 'src/core/server';
+
+const KIBANA_DATA_ROOT = 'kibana';
+
+export interface Profile<T extends UserData> {
+  uid: string;
+  user: User;
+  data: T;
+}
+
+export interface User {
```

Comment:
__question:__ Is it supposed to have a different shape than our existing `User` interface?

Reply:
were still defining exact shape but can extend from base interface